helix-term/src/file_info.rs

```diff
@@ -0,0 +1,390 @@
+use std::path::PathBuf;
+
+use tui::text::{Span, Spans};
+
+use crate::ui::menu::Item;
+
+pub struct FileInfoData {
+    pub root_path: PathBuf,
+    pub show_icons: bool,
+}
+
+pub struct FileInfo {
+    path: PathBuf,
+    icon_character: char,
+}
+
+impl FileInfo {
+    pub fn new(path: PathBuf, icon_character: char) -> Self {
+        Self {
+            path,
+            icon_character,
+        }
+    }
+
+    pub fn icon_character(&self) -> char {
+        self.icon_character
+    }
+
+    pub fn path(&self) -> &PathBuf {
+        &self.path
+    }
+}
+
+impl Item for FileInfo {
+    /// Root prefix to strip.
+    type Data = FileInfoData;
+
+    fn label(&self, data: &Self::Data) -> Spans {
+        let mut result = vec![Span::raw(format!(
+            "  {}",
```

Comment:
If this isn't done then the icon overwrites the first character during render, and I'm not sure why. Any ideas?

Reply:
I believe that's a bug that should be solved by #3839